R/check_year_format.R

```diff
@@ -27,18 +27,18 @@ check_year_format <- function(year, format = "fyyear") {
 
   if (format == "fyyear") {
     if (any(as.integer(first_part) + 1L != as.integer(second_part))) {
```

Comment:
I noticed that if you give a vector like `c("1819", NA, "1920")` the function will abort but won't print a specific error message acknowledging that

Reply:
I think in this situation it should just throw an error?

Could modify `check_year_type` to do that?

Reply:
I think I've addressed this now.

I also noticed some tests were missing from one of the functions, so I added those and then had to write code to pass them properly!